doc/sphinx/practical-tools/utilities.rst

```diff
@@ -124,22 +124,28 @@ avoid putting your code into subdirectories such as "src" unless you want
 that to be part of the :term:`logical path` in Coq.
 
 For example, here's a minimal _CoqProject file for the `MyProject` package
-(the logical name of the package), which has two script files `File1.v`
-and `SubDir/File2.v` whose logical paths
-are `MyProject.File1` and `MyProject.SubDir.File2`.  `_CoqProject` should be
-in the top-level directory of your project.  We recommend naming that directory
-`MyProject`, the same as the logical name::
+(the logical name of the package), which includes all the ``.v`` files (and
+other file types) in the current directory and its subdirectories:
+
+  -R . MyPackage
+  .
+
+You can list multiple directories if you wish.
+
+In addition, you can list individual files, for example for the two script files
+`File1.v` and `SubDir/File2.v` whose logical paths are `MyProject.File1` and
+`MyProject.SubDir.File2`.  We recommend naming that directory `MyProject`, the
```

Comment:
Who are we to tell the user how they should name their directories?

Moreover, this is a recommendation that might be painful to follow for users. For example, the current directory might well be the name of a cloned git repository (e.g., `coq-foo`) while the logical path is quite different (e.g., `Foo`). Sure, users could always rename their checkout directory to match it against the logical path, but why should they care and why do we?

Reply:
> Who are we to tell the user how they should name their directories?

There's nothing wrong _per se_ in giving advice, indeed advice can often be quite helpful.  Of course it has to be good advice.  Users will do what they wish.

> For example, the current directory might well be the name of a cloned git repository (e.g., coq-foo) while the logical path is quite different (e.g., Foo).

A good point. I had not considered that case when I wrote that section.  For a single package project, the IDE gets the `-R/-Q` information from `_CoqProject` and passes that to Coq, so `coq-foo` is fine.  But the IDE only uses a single `_CoqProject`, which isn't sufficient for projects that have multiple uninstalled packages, e.g. because you're editing multiple packages at once.  Packages that you're editing should not be in the `user-contrib` directory, so you need to use `$COQPATH` to point to them.  COQPATH uses the directory name as the first part of the logical name, hence the recommendation.

I have a proposal that will allow handling both your `coq-foo` use case and the multipackage case as well as a few others we currently don't handle well.  I planned to bring that up shortly after completing this PR.  Whether that proposal is implemented or not, I'd love to improve this section to address your concerns.  I think that would work best as a separate discussion rather than in this PR.  It's really a different topic from allowing directory names in `_CoqProject`.  Note that I'm not really adding new recommendations in this PR, but rather moving them from one spot to another.  Is this a reasonable approach?

Reply:
> Moreover, this is a recommendation that might be painful to follow for users.

Calling this "painful" seems an exaggerated.  git doesn't care about the name of the directory containing a cloned git repository.

Reply:
While I agree that the advice is essentially the existing one, I agree it violates the most common conventions.

It's also pretty confusing — if you have a folder containing the suggested `_CoqProject`, `File1.v`, and `SubDir/File2.v`, then the name of that folder is technically irrelevant in this scenario.

> For a single package project, the IDE gets the -R/-Q information from _CoqProject and passes it to Coq, so coq-foo is fine.

Dashes are common in project nams, but they're illegal in logical names — you can't use `-Q . my-package`, because `Require my-package.foo` does not parse correctly.

Reply:
Let's refine the recommendations very soon, but not in this PR.